src/firejail/firejail.h

```diff
@@ -337,6 +337,7 @@ extern int arg_writable_run_user;	// writable /run/user
 extern int arg_writable_var_log; // writable /var/log
 extern int arg_appimage;	// appimage
 extern int arg_apparmor;	// apparmor
+extern char apparmor_profile[30];	// apparmor profile
```

Comment:
```suggestion
char *char apparmor_profile = NULL;	// apparmor profile
```

Why so short?  The profile could live in a long path within the user's home
directory.

Also, the size of argv is already checked (see `MAX_ARG_LEN`), and there is no
such restriction on the paths of `--hosts-file=` nor `--netfilter=`.

So do it similarly to this:

```C
char *arg_netfilter_file = NULL;			// netfilter file
```

Reply:
OK, sorry. I've made this patch for my laptop, and I mostly use custom profile names. I forgot that they can be big sometimes.

Reply:
> Why so short? The profile could live in a long path within the user's home directory.

**apparmor_profile** doesn't store a path to the profile, it stores the name of the profile that is already loaded to the kernel. It uses **aa_stack_onexec** that simply schedules a domain transition -- it doesn't load any new profiles.

Reply:
@ChrysoliteAzalea [on Jul 26](https://github.com/netblue30/firejail/pull/5274#discussion_r930354716):

> > Why so short? The profile could live in a long path within the user's home
> > directory.
> 
> **apparmor_profile** doesn't store a path to the profile, it stores the name
> of the profile that is already loaded to the kernel. It uses
> **aa_stack_onexec** that simply schedules a domain transition -- it doesn't
> load any new profiles.

Ah I see, sorry.